### DISCUSSION TITLE
What is a good high-level Assembler?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#118599] [Programming Help]
### Body

Hi,

I am learning Assembly language for the X86 architecture, specifically 32 bit, but later on I got a book for 64 bit Assembly. Can you recommend a high-level Assembler I could use?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by CopperEagle** [comment]
I would recommend just going for 64 bit, since 32 bit is getting increasingly rare nowadays. In any case, `nasm` should have you covered.

Just found this good ![gist](https://gist.github.com/yellowbyte/d91da3c3b0bc3ee6d1d1ac5327b1b4b2) that helps beginners in Assembly on how to compile, link and gives some templates. It is specifically for Linux, though.

**Reply by ebndev** [reply]
Hey @CopperEagle!

I've been seeing some of your comments around Programming Help and just wanted to say thanks for being an active member in the community 😁

**Reply by CopperEagle** [reply]
@ebndev Aww, thanks a lot for the compliment! 😁
